fastestimator/summary/system.py

```diff
@@ -34,8 +34,8 @@ class System:
 
     def __init__(self,
                  network: BaseNetwork,
-                 mode: str = "train",
-                 num_devices: int = get_num_devices(),
+                 mode: Optional[str] = None,
```

Comment:
why is mode optional? shouldn't the system always be in one mode or another?

Reply:
warmup is using None as mode.  there's an if condition that needs it.

Reply:
ah, ok. we should update the class-var hint then (line 23) with that explanation

Reply:
done